spec/models/manageiq/providers/kubernetes/container_manager/metrics_capture/capture_context_spec.rb

```diff
@@ -0,0 +1,90 @@
+      'gBbHTTagOHwhrZmMSgwfwpzPAiFij42xF61HCK_Z'\
+      'LrDEZ7_HDfHAXRa0pHEbh4D2IQ'
+
+    @ems = FactoryGirl.create(
+      :ems_openshift,
+      :name                      => 'QaLab',
+      :connection_configurations => [{:endpoint       => {:role     => :default,
+                                                          :hostname => hostname,
+                                                          :port     => "8443"},
+                                      :authentication => {:role     => :bearer,
+                                                          :auth_key => token,
+                                                          :userid   => "_"}},
+                                     {:endpoint       => {:role     => :hawkular,
+                                                          :hostname => hostname_hawkular,
+                                                          :port     => "443"},
+                                      :authentication => {:role     => :hawkular,
+                                                          :auth_key => token,
+                                                          :userid   => "_"}}])
+  end
+
+  it "will read hawkular status (does not require authentication)" do
+    VCR.use_cassette("#{described_class.name.underscore}_status") do # , :record => :new_episodes) do
+      EmsRefresh.refresh(@ems)
+      context = ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::CaptureContext.new(
+        @ems.container_nodes.last, 1_466_505_130, 1_466_508_726, 20)
+
+      metrics = {"MetricsService"         => "STARTED",
+                 "Implementation-Version" => "0.8.1.Final-redhat-1",
+                 "Built-From-Git-SHA1"    => "f8718499684ef4b9beb456667fa7da4786e7fadd"}
+
+      data = context.hawkular_client.http_get('/status')
+
+      expect(data).to eq(metrics)
+    end
+  end
+
+  it "will read hawkular system metrics (requires authentication)" do
+    VCR.use_cassette("#{described_class.name.underscore}_system_metrics") do # , :record => :new_episodes) do
+      EmsRefresh.refresh(@ems)
+      context = ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::CaptureContext.new(
+        @ems.container_nodes.last, 1_466_505_130, 1_466_508_726, 20)
+
+      data = context.hawkular_client.http_get('/metrics')
+
+      expect(data).to be_a_kind_of(Array)
+      expect(data.length).to eq(14)
+      expect(data[13]['type']).to be
+    end
+  end
+
+  it "will read hawkular node metrics" do
+    VCR.use_cassette("#{described_class.name.underscore}_metrics") do # , :record => :new_episodes) do
+      EmsRefresh.refresh(@ems)
+      context = ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::CaptureContext.new(
+        @ems.container_nodes.first, 1_466_505_130, 1_466_508_726, 20)
+
+      data = context.collect_metrics
+
+      expect(data).to be_a_kind_of(Array)
+      expect(data.length).to eq(566)
+      expect(data[565]['start']).to be
```

Comment:
@cben Thanks, added a note about setting up providers, change hostname, and use explicit time :+1:  .to be is a way to ask for exist